akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -292,17 +292,14 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
     parallelism:       Int               = 1,
     log:               LoggingAdapter    = system.log)(implicit fm: Materializer): Future[ServerBinding] = {
     val http2enabled = settings.previewServerSettings.enableHttp2
-    if (http2enabled && connectionContext.isSecure && connectionContext.http2 != Never) {
-      log.debug("Binding server using HTTP/2...")
-      Http2Shadow.bindAndHandleAsync(handler, interface, port, connectionContext, settings, parallelism, log)(fm)
-    } else if (http2enabled && !connectionContext.isSecure && connectionContext.http2 == Always) {
-      // We do not support HTTP/2 negotiation for insecure connections (h2c), https://github.com/akka/akka-http/issues/1966
+    if (http2enabled && connectionContext.http2 != Never) {
```

Comment:
This part of the change is nontrivial because `connectionContext.http2` by default is `Negotiated`, so after this change the code path at https://github.com/akka/akka-http/pull/2085/files#diff-f0a57b2d6e8f16da4ca0286a011211e1R109 will be taken, which might have a performance impact.

Given the limited support for h2c `Upgrade`, I think we should switch the default to `Never` for insecure connections and `Negotiated` only for secure connections.

WDYT?

Reply:
Sounds like it makes sense.